### PR TITLE
Create Storage Blob Client with list blobs API

### PR DESCRIPTION
**Pull request by vcolin7**
The [first commit](https://github.com/Azure/azure-sdk-for-android/pull/174/commits/34c45f0c535e36f288b05530cd0c349631f40302) is models copied from Java sdk with small update to use threeten time type.
The [second commit](https://github.com/Azure/azure-sdk-for-android/pull/174/commits/989fb32d7fa4e1ce01bcd61ebd00a9c825d426ce) the one we really care about, which adds `StorageBlobClient` with list blobs API.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobClient.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+     *
+     * @param pageId Identifies the portion of the list to be returned.
+     * @param containerName The container name.
+     * @param options The page options.
+     * @param callback Callback that receives the retrieved blob list.
+     */
+    public void getBlobsInPage(String pageId,
+                               String containerName,
+                               ListBlobsOptions options,
+                               ServiceCallBack<List<BlobItem>> callback) {
```
Comment (anuchandy) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobClient.java]: Info: We've both sync and async methods. I found call-back based async is natural to use with android framework APIs such as pagination.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                prefix,
+                pageId,
+                maxResults,
+                this.serializerAdapter.serializeList(include, SerializerAdapter.CollectionFormat.CSV),
+                timeout,
+                XMS_VERSION,
+                requestId,
+                resType,
+                comp
+                /*context*/));
```
Comment (anuchandy) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java]: Info: commented context will be removed, we haven't encountered need for pipeline context so far in mobile, so putting it in back log.

**Comment by bsiegel** [comment]
Just had one suggestion, otherwise this LGTM.

File: sdk/core/azure-core/src/main/java/com/azure/android/core/http/ServiceCallBack.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+// Licensed under the MIT License.
+
+package com.azure.android.core.http;
+
+/**
+ * Callback to receive a service operation result.
+ *
+ * @param <T> The type of the result.
+ */
+public interface ServiceCallBack<T> {
```
[the remark this answers is not in the document]
Comment (bsiegel) [reply sdk/core/azure-core/src/main/java/com/azure/android/core/http/ServiceCallBack.java]: If "callback" is common for Android e.g. in OkHttp, Maybe just call this class `Callback`?

[the remark this answers is not in the document]
Comment (anuchandy) [reply sdk/core/azure-core/src/main/java/com/azure/android/core/http/ServiceCallBack.java]: One of the goal while designing our serviceClient was - the support to take an existing OkHttpClient as the base. If I remember correctly we talked about cases where customers want to share an OkHttpClient his app uses for different purposes. In such cases, yes there are chances to use these two callbacks. That being said, I like `com.azure.android.core.http.Callback`, given its a pattern we generally see - okhttp3.Callback, retrofit2.Callback

**Comment by vcolin7** [comment]
@bsiegel @anuchandy Renamed the ServiceCallBack to Callback and solved a small merge conflict.

**Pull request merged**